packages/components/app/styles/components/badge.scss

```diff
@@ -144,3 +144,12 @@ $size-props: (
     }
   }
 }
+
+.hds-form-label,
+.hds-form-legend {
+  .hds-badge {
+    display: inline-block;
+    vertical-align: baseline;
+    margin-left: 4px;
```

Comment:
[nitpick(ux)] font-weight is explicitly called out for the optional indicator, but for required it's taking on the font-weight from the label. The font-weight of the required badge should be `--token-typography-font-weight-medium`.

Reply:
I agree with Heather here. We may also have more indicators beyond required and optional with longer text, which would make the label even more challenging to read.

Reply:
@heatherlarsen the problem is because we (I) never explicitly set the `font-weight` on the `Badge` component.
I've opened a PR to fix this: https://github.com/hashicorp/design-system/pull/426
/cc @alex-ju